Improve swipe gesture to reveal year scroller

tehapo opened the pull request:
Fixes #190 

Some notes for review.
* Only way to really review this is to simply try this out with an Android and/or iOS device. If your device has too large screen, you might need to tweak the `iron-media-query` in `vaadin-date-picker-overlay` and increase the `min-width`.
* This PR improves three things.
  * A costly repaint is avoided by adding the `will-change` CSS property to the icon that gets rotated.
  * The `on-track` is moved one level up in the DOM to actually track also when swiping on top of the year scroller to hide it.
  * Handling the `track` state is rewritten and should now feel more natural and also support a fast fling instead of just a deliberate and slow swipe.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/vaadin/vaadin-date-picker/242)
<!-- Reviewable:end -->

samiheikki commented:
Tested on ios and android, seems to work really nice

---

Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[vaadin-date-picker-overlay.html, line 523 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/242#-KSFkCVRCdVuGR8Vq6ax:-KSFkCVS9dGhhZb99v2j:b-dmrbxn) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/1c519e66cd66e2f7cf40776357db47508ceaacfe/vaadin-date-picker-overlay.html#L523)):*
> ```HTML
> 
>       _handleTrack: function(e) {
>         if (Math.abs(e.detail.dx) < 10 || Math.abs(e.detail.ddy) > 10) {
> ```

What does magic number `10` mean in here? have no idea :)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/242#-:-KSFjnt1nixaIVGZ3rkL:bjg90a9)*
<!-- Sent from Reviewable.io -->

tehapo commented:
Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[vaadin-date-picker-overlay.html, line 523 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/242#-KSFkCVRCdVuGR8Vq6ax:-KSKbCin1RrxlArK5Ooq:b-kxcnv8) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/1c519e66cd66e2f7cf40776357db47508ceaacfe/vaadin-date-picker-overlay.html#L523)):*
<details><summary><i>Previously, samiheikki (Sami Suo-Heikki) wrote…</i></summary>
> What does magic number `10` mean in here? have no idea :)
</details>
Moved and clarified some related comments. Would this be better?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/242)*
<!-- Sent from Reviewable.io -->

samiheikki commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/242#-:-KSKngoQ-cnaq7qz602D:bnfp4nl)*
<!-- Sent from Reviewable.io -->

manolo commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/242#-:-KSKxAQi9XABatuDTtl5:bnfp4nl)*
<!-- Sent from Reviewable.io -->

tomivirkki commented:
For some reason the year scroller isn't scrollable (on iPad 2) with this change... if that's not an issue then <img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---







*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/242#-:-KSL1uthp8HzHYxJ74sB:bp93d9q)*
<!-- Sent from Reviewable.io -->

manolo commented:
was it before ? 

---

Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/242#-:-KSL6xHXkktXJpsxDinA:buqo5yh)*
<!-- Sent from Reviewable.io -->

tomivirkki commented:
Yes it was

---







*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/242#-:-KSL77UB94cOuftk8web:b6pwp82)*
<!-- Sent from Reviewable.io -->

manolo commented:
So probably we shouldn't break that. Could you @tehapo take a look before merging this to figure out what is broken?

---

Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/242#-:-KSLCC_HG95vhnI7ePE4:b4pviaz)*
<!-- Sent from Reviewable.io -->

tehapo commented:
Oh, good catch. I'll take a look at it.

---

Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/242#-:-KSLF6GQCI91vwn_pnjP:bar3e5h)*
<!-- Sent from Reviewable.io -->

tehapo commented:
Fixed. And also rebased on top of master.

---

Review status: 0 of 1 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/242#-:-KSLzLVL5SFIg6r4O3aH:b-m3aee8)*
<!-- Sent from Reviewable.io -->

tomivirkki commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: 0 of 1 files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/242#-:-KSMBp96EyWo8QVdR7_a:bnfp4nl)*
<!-- Sent from Reviewable.io -->

tehapo commented:
Fixed linting error.

---

Review status: 0 of 1 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/242#-:-KSZvy7fw8rTugxr9aXF:b-lmd39k)*
<!-- Sent from Reviewable.io -->

samiheikki commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/242#-:-KS_PL4bWG4LB2c-r5R2:bnfp4nl)*
<!-- Sent from Reviewable.io -->
